runtime/compiler/optimizer/InlinerTempForJ9.cpp

```diff
@@ -1120,6 +1191,36 @@ TR_J9InlinerPolicy::createUnsafePutWithOffset(TR::ResolvedMethodSymbol *calleeSy
    // Preserve null check on the unsafe object
    TR::TransformUtil::separateNullCheck(comp(), callNodeTreeTop, tracer()->debugLevel());
 
+
+   bool conversionNeeded = comp()->fe()->dataTypeForLoadOrStore(type) != type;
+
+   // If we are not able to get javaLangClass it is still inefficient to put direct Access far
+   // So in that case we will generate lowTagCmpTest to branch to indirect access if true
+   TR_OpaqueClassBlock *javaLangClass = comp()->getClassClassPointer(/* isVettedForAOT = */ true);
+   bool needNotLowTagged = javaLangClass != NULL  || conversionNeeded ;
+
+   int length;
+   const char *objTypeSig = unsafeCall->getChild(1)->getSymbolReference()->getTypeSignature(length);
+
+   // There are two cases where IL for type checks (array, class, lowtag) needs to be generated:
+   // 1.) Object type is unknown at compile time
+   // 2.) Object is known at compile time to be a java/lang/Class object (to match JNI implementation)
+   bool checksNeeded = objTypeSig == NULL || strstr(objTypeSig, "Ljava/lang/Object") || strstr(objTypeSig, "Ljava/lang/Class");
```

Comment:
general comment on coding style:  use appropriate brackets for better readability.

Reply:
have you considered the possibility of **objTypeSig** being something like  **Ljava/lang/ObjectFoo** or **Ljava/lang/ClassBar**?  how about **objTypeSig** being **[Ljava/lang/Object**?